src/crimson/os/seastore/async_cleaner.h

```diff
@@ -1320,7 +1320,9 @@ class SegmentCleaner : public SegmentProvider, public AsyncCleaner {
     auto aratio = segments.get_available_ratio();
     auto rratio = get_reclaim_ratio();
     return (
-      (aratio < config.available_ratio_hard_limit) ||
+      // make sure the cleaner will start when try_reserve fails
+      (aratio < config.available_ratio_hard_limit +
+       1.0 / segments.get_num_segments()) ||
```

Comment:
I'm not sure why the original condition won't work, and why this should work?

Seems it enforces the `available_ratio_hard_limit` to be always larger than 1 available segment, but I don't see the connection to the try_reserve issue.

Reply:
Yeah, I agree that cleaning cold tier should be enforced when `try_reserve` in the cold tier is failed, and it's not related to this PR. Also see https://github.com/ceph/ceph/pull/48691#discussion_r1027644146